.github/workflows/build-test.yml

```diff
@@ -217,6 +217,130 @@ jobs:
           path: ./dev/**  # use wildcard pattern to preserve dir structure of uploaded files
           name: benchmark-perf
 
+  # Run examples with dhat-rs in order to collect memory heap size metrics. These
+  # metrics will then be charted over time. See utils/memory_bench/README.md for
+  # more information.
+
+  # dhat-rs:
+  #   https://github.com/nnethercote/dhat-rs
+  # Benchmarking action (forked):
+  #   https://github.com/gregtatum/github-action-benchmark
+  # The memory data is collected in:
+  #   benchmarks/memory/{os}/{component_path}/output.ndjson
+  # The full data report is stored in:
+  #   benchmarks/memory/{os}/{component_path}/{example}-dhat-heap.json
+
+  memory:
+    strategy:
+      fail-fast: false
+      # Create a matrix of all platforms, and all components. Each job then can run
+      # multiple examples in that job. The examples are defined as a space separated
+      # list of the name of the examples. The examples are assumed to be in the
+      # examples folder.
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+        example: [
+          { component: components/datetime, examples: "work_log" },
+          { component: components/locid, examples: "syntatically_canonicalize_locales filter_langids" },
+          { component: components/plurals, examples: "unread_emails elevator_floors" },
+          { component: components/uniset, examples: unicode_bmp_blocks_selector },
+          # Important! When adding new components, ensure that dhat is added as
+          # dev dependency, or else the job will fail.
+        ]
+
+    runs-on: ${{ matrix.os }}
+
+    # This job waits on the full build, even though it doesn't need anything from it.
+    # This way it is only triggered if the build is successful. The CI could be made
+    # a little bit faster by starting this job immediately, as it's fully independent.
+    needs: [build]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Nightly is only needed for the `--profile bench` line. Once that is stabilized,
+      # then this line is no longer needed.
+      - name: Install nightly toolchain
+        run: rustup toolchain install nightly
+
+      - name: Install cargo-make
+        uses: davidB/rust-cargo-make@v1
+        with:
+          version: latest
+
+      - name: Run the example with dhat-rs to collect memory information
+        run: |
+          cargo run --package memory_bench -- ${{ matrix.os }} ${{ matrix.example.component }} ${{ matrix.example.examples }}
+
+      # Benchmarking & dashboards job > (unmerged PR only) Convert benchmark output into
+      # dashboard HTML in a commit of a branch of the local repo.
+
+      - name: Store benchmark result & create dashboard (unmerged PR only)
+        # any action that is not a merge to master implies unfinished PR
+        if: github.event_name != 'push' || github.ref != 'refs/heads/master'
+        uses: gregtatum/github-action-benchmark@v1
```

Comment:
Is this action pre-existing, or did you create it for icu4x?

Reply:
This is the forked version of the action used by the performance benchmark, it adds support for ndjson.

[https://github.com/gregtatum/github-action-benchmark](https://github.com/gregtatum/github-action-benchmark/commit/a07750e71f9251c3a5e2da2dab921bd479ff71ba)